src/dmd/dmodule.d

```diff
@@ -1355,3 +1382,234 @@ struct ModuleDeclaration
+            }
+            index += 1 + info.depth;
+        }
+        return index;
+    }
+
+    if (matchNodes.dim == 0)
+    {
+        for (size_t packageIndex = 0; packageIndex < global.params.includeMatchStrings.dim; packageIndex++)
+        {
+            auto matchString = global.params.includeMatchStrings[packageIndex];
+            auto depth = parseMatchStringDepth(matchString);
+            auto entryIndex = findSortedIndexToAddForDepth(depth);
+            matchNodes.split(entryIndex, depth + 1);
+            parseMatchString(matchString, &matchNodes[entryIndex], depth);
+            // if at least 1 "include pattern" is given, then it is assumed the
+            // user only wants to include modules that were explicitly given, which
+            // changes the default behavior from inclusion to exclusion.
+            if (includeByDefault && !matchNodes[entryIndex].isExclude)
+            {
+                //fprintf(global.stdmsg, "Matcher: found 'include pattern', switching default behavior to exclusion\n");
+                includeByDefault = false;
+            }
+        }
+
+        // Add the default 1 depth matchers
+        MatcherNode[8] defaultDepth1MatchNodes = [
+            MatcherNode(true, 1), MatcherNode(Id.std),
+            MatcherNode(true, 1), MatcherNode(Id.core),
+            MatcherNode(true, 1), MatcherNode(Id.etc),
+            MatcherNode(true, 1), MatcherNode(Id.object),
+        ];
+        {
+            auto index = findSortedIndexToAddForDepth(1);
+            matchNodes.split(index, defaultDepth1MatchNodes.length);
+            matchNodes.data[index .. index + defaultDepth1MatchNodes.length] = defaultDepth1MatchNodes[];
+        }
+    }
+}
+private bool endOfPattern(char c) { return c == ',' || c == '\0'; }
+private ushort parseMatchStringDepth(const(char)* matchString)
+{
+    if (matchString[0] == '-')
+        matchString++;
+
+    // handle special case
+    if (matchString[0] == '.' && endOfPattern(matchString[1]))
+        return 0;
+
+    ushort depth = 1;
+    for (;; matchString++)
+    {
+        auto c = *matchString;
+        if (c == '.')
+            depth++;
+        if(endOfPattern(c))
+            return depth;
+    }
+}
+private void parseMatchString(const(char)* matchString, MatcherNode* dst, ushort depth)
+{
+    bool isExclude = false;
+    if (matchString[0] == '-')
+    {
+        isExclude = true;
+        matchString++;
+    }
+
+    *dst = MatcherNode(isExclude, depth);
+    dst++;
+
+    // Create and add identifiers for each component in the matchString
+    if (depth > 0)
+    {
+        static Identifier getID(const(char)* start, const(char)* limit)
+        {
+            if (limit == start)
+                // empty patterns should be impossible because this is checked earlier
+                assert(0, "identifer cannot be empty");
+
+            auto length = limit - start;
+            // TODO: not sure if this the right logic to create the identifier?
```

Comment:
Was hoping someone could confirm this is the right way to create an identifier in this instance.